presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/MemoBasedLookup.java

```diff
@@ -60,4 +65,14 @@ public PlanNodeStatsEstimate getStats(Session session, Map<Symbol, Type> types,
                 session,
                 types));
     }
+
+    @Override
+    public PlanNodeCostEstimate getCost(Session session, Map<Symbol, Type> types, PlanNode planNode)
+    {
+        return costs.computeIfAbsent(resolve(planNode), node -> costCalculator.calculateCost(
```

Comment:
We need to be able to get the cumulative cost of the plan, and not just the added cost of each node. Maybe that should be what's returned by the cost calculator.  I don't see a use case where getting the plan cost instead of the node cost would be a problem.  But if not, we need two methods for Lookup getNodeCost() and getPlanCost().